File: src/main/java/com/senderman/lastkatkabot/command/user/GetInfo.java
package com.senderman.lastkatkabot.command.user;

import com.annimon.tgbotsmodule.commands.context.MessageContext;
import com.senderman.lastkatkabot.command.CommandExecutor;
import org.springframework.stereotype.Component;

@Component
public class GetInfo implements CommandExecutor {

    public GetInfo() {
    }

    @Override
    public String getTrigger() {
        return "/getinfo";
    }

    @Override
    public String getDescription() {
        return "(reply) инфа о сообщении в формате JSON";
    }

    @Override
    public void execute(MessageContext ctx) {
        var message = ctx.message();
        if (!message.isReply()) {
            ctx.replyToMessage("Для использования команды, отправьте ее в ответ на нужное сообщение!").callAsync(ctx.sender);
            return;
        }

        var replyInfo = message.getReplyToMessage()
                .toString()
                .replaceAll("\\w+=null,?\\s*", "")
                .replaceAll("=([\\w\\d]+)", "=<code>$1</code>");

        ctx.reply(replyInfo).callAsync(ctx.sender);
    }
}

File: src/main/java/com/senderman/lastkatkabot/usercommands/Getinfo.kt
package com.senderman.lastkatkabot.usercommands

import com.senderman.CommandExecutor
import com.senderman.lastkatkabot.LastkatkaBotHandler
import org.telegram.telegrambots.meta.api.objects.Message

class Getinfo constructor(val handler: LastkatkaBotHandler) : CommandExecutor {
    override val command: String
        get() = "/getinfo"
    override val desc: String
        get() = "(reply) инфа о сообщении в формате JSON"

    override fun execute(message: Message) {
        if (!message.isReply) return

        val replacements = mapOf(
                "[ ,]*\\w+='?null'?" to "",
                "(\\w*[iI]d=)(-?\\d+)" to "$1<code>$2</code>",
                "([{,])" to "$1\n",
                "(})" to "\n$1",
                "(=)" to " $1 "
        )
        var text = message.replyToMessage.toString()
        for ((old, new) in replacements) text = text.replace(old.toRegex(), new)
        handler.sendMessage(message.chatId, text)
    }
}